core/shared/src/main/scala/magnolia.scala

```diff
@@ -577,11 +595,21 @@ object Magnolia {
 
         val subtypesVal = c.freshName(TermName("subtypes"))
         val typeclasses = for (subType <- subtypes) yield {
+          val anns = annotationsOf(subType.typeSymbol)
           val path = CoproductType(genericType.toString)
           val frame = stack.Frame(path, resultType, assignedName)
-          subType -> stack.recurse(frame, appliedType(typeConstructor, subType), shouldCache) {
-            typeclassTree(subType, typeConstructor, termNames.ERROR)
-          }.fold(error, identity)
+          val isRuntimeBound = anns.exists {
+            case Apply(Select(New(typeTree), _), _) if isRuntimeBoundTypeclassAnnotation(typeTree) => true
+            case _ => false
+          }
+
+          subType -> (if (isRuntimeBound) {
+            q"${c.prefix}.bind[$subType]"
```

Comment:
Can we use `fallback` instead of `bind`? I didn't ever use the `fallback` method, but I think it serves the same purpose.

Reply:
I did not notice before that there is `fallback`. I think it's a good idea to reuse it for this use case too, then instead of talking about "runtime bound" it could be "forced fallback".